examples/plot_meg_example.py

```diff
@@ -0,0 +1,184 @@
+        that are parallel (tangential) to the cortical surface. If loose
+        is 0 then the solution is computed with fixed orientation.
+        If loose is 1, it corresponds to free orientations.
+        The default value ('auto') is set to 0.2 for surface-oriented source
+        space and set to 1.0 for volumic or discrete source space.
+    depth : None | float in [0, 1]
+        Depth weighting coefficients. If None, no depth weighting is performed.
+
+    Returns
+    -------
+    stc : instance of SourceEstimate
+        The source estimates.
+    """
+    # Import the necessary private functions
+    from mne.inverse_sparse.mxne_inverse import \
+        (_prepare_gain, is_fixed_orient,
+         _reapply_source_weighting, _make_sparse_stc)
+
+    all_ch_names = evoked.ch_names
+
+    # Handle depth weighting and whitening (here is no weights)
+    forward, gain, gain_info, whitener, source_weighting, mask = _prepare_gain(
+        forward, evoked.info, noise_cov, pca=False, depth=depth,
+        loose=loose, weights=None, weights_min=None, rank=None)
+
+    # Select channels of interest
+    sel = [all_ch_names.index(name) for name in gain_info['ch_names']]
+    M = evoked.data[sel]
+
+    # Whiten data
+    M = np.dot(whitener, M)
+
+    n_orient = 1 if is_fixed_orient(forward) else 3
+
+    X, active_set = solver(M, gain, n_orient)
+    X = _reapply_source_weighting(X, source_weighting, active_set)
+
+    stc = _make_sparse_stc(X, active_set, forward, tmin=evoked.times[0],
+                           tstep=1. / evoked.info['sfreq'])
+
+    return stc
+
+
+###############################################################################
+# Define your solver
+
+def solver(y_train, X_train, n_orient):
+    """Run L2 penalized regression and keep 10 strongest locations.
+
+    Parameters
+    ----------
+    M : array, shape (n_channels, n_times)
+        The whitened data.
+    G : array, shape (n_channels, n_dipoles)
+        The gain matrix a.k.a. the forward operator. The number of locations
+        is n_dipoles / n_orient. n_orient will be 1 for a fixed orientation
+        constraint or 3 when using a free orientation model.
+    n_orient : int
+        Can be 1 or 3 depending if one works with fixed or free orientations.
+        If n_orient is 3, then ``G[:, 2::3]`` corresponds to the dipoles that
+        are normal to the cortex.
+
+    Returns
+    -------
+    X : array, (n_active_dipoles, n_times)
+        The time series of the dipoles in the active set.
+    active_set : array (n_dipoles)
+        Array of bool. Entry j is True if dipole j is in the active set.
+        We have ``X_full[active_set] == X`` where X_full is the full X matrix
+        such that ``M = G X_full``.
+    """
+    n_times = y_train.shape[1]
+    idx_max = np.argmax(np.sum(y_train ** 2, axis=0))
+    y_train = y_train[:, idx_max]
+
+    n_samples, n_features = X_train.shape
+    alpha_max = (np.abs(X_train.T @ y_train)).max() / n_samples
```

Comment:
you compute twice alpha_max. I don't get how you can scale X back to its origin unit in nAm with the code above.